mlflow/server/js/src/shared/web-shared/model-trace-explorer/ModelTraceHeaderDetails.tsx

```diff
@@ -0,0 +1,217 @@
+
+  const handleClick = () => {
+    navigator.clipboard.writeText(value);
+    onCopy();
+  };
+
+  return (
+    <div
+      css={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        flexDirection: 'row',
+        gap: theme.spacing.sm,
+      }}
+    >
+      <Typography.Text size="md" color="secondary">
+        {label}
+      </Typography.Text>
+      <Tooltip componentId={getComponentId(tagKey)} content={value} maxWidth={400}>
+        <Tag componentId={getComponentId(tagKey)} color={color} onClick={handleClick} css={{ cursor: 'pointer' }}>
+          {getTruncatedLabel(value)}
+        </Tag>
+      </Tooltip>
+    </div>
+  );
+};
+
+export const ModelTraceHeaderDetails = ({ modelTrace }: { modelTrace: ModelTrace }) => {
+  const intl = useIntl();
+  const { theme } = useDesignSystemTheme();
+  const [showNotification, setShowNotification] = useState(false);
+
+  const tags = Object.entries(modelTrace.info.tags ?? {}).filter(([key]) => isUserFacingTag(key));
+
+  const modelTraceId = getModelTraceId(modelTrace);
+
+  const tokenUsage = parseJSONSafe(
+    (modelTrace.info as ModelTraceInfoV3)?.trace_metadata?.['mlflow.trace.tokenUsage'] ?? '{}',
+  );
+
+  const totalTokens = useMemo(() => tokenUsage?.total_tokens, [tokenUsage]);
+
+  const spans = modelTrace?.trace_data?.spans ?? modelTrace?.data?.spans;
+
+  const calculateLatency = useCallback((): string | undefined => {
+    const rootSpan = spans?.find((span) => !getModelTraceSpanParentId(span));
+
+    if (rootSpan) {
+      const startTime = getModelTraceSpanStartTime(rootSpan);
+      const endTime = getModelTraceSpanEndTime(rootSpan);
+
+      if (startTime && endTime && endTime > startTime) {
+        return `${((endTime - startTime) / 1000000).toFixed(1)}s`;
+      }
+    }
+
+    return undefined;
+  }, [spans]);
+
+  const latency = calculateLatency();
+
+  const idLabel = intl.formatMessage({
+    defaultMessage: 'ID',
+    description: 'Label for the ID section',
+  });
+
+  const tokenCountLabel = intl.formatMessage({
+    defaultMessage: 'Token count',
+    description: 'Label for the token count section',
+  });
+
+  const latencyLabel = intl.formatMessage({
+    defaultMessage: 'Latency',
+    description: 'Label for the latency section',
+  });
+
+  const tagsLabel = intl.formatMessage({ defaultMessage: 'Tags', description: 'Label for the tags section' });
+
+  const notificationSuccessMessage = intl.formatMessage({
+    defaultMessage: 'Copied to clipboard',
+    description: 'Success message for the notification',
+  });
+
+  const getComponentId = useCallback((key: string) => `${BASE_TAG_COMPONENT_ID}.tag-${key}`, []);
+
+  const handleTagClick = (text: string) => {
+    navigator.clipboard.writeText(text);
+  };
+
+  const getTruncatedLabel = (label: string) => truncateToFirstLineWithMaxLength(label, 40);
+
+  const handleCopy = useCallback(() => {
+    setShowNotification(true);
```

Comment:
future consideration: we use different pattern in DB products (tooltip over the clicked element instead of a toast), but corresponding code is not ported to OSS yet. I propose to leave it as is for now and move to ported clipboard utils&component as a followup